crates/catalog/sql/src/catalog.rs

```diff
@@ -476,51 +484,311 @@ impl Catalog for SqlCatalog {
+             WHERE {CATALOG_FIELD_CATALOG_NAME} = ? 
+             AND {CATALOG_FIELD_TABLE_NAMESPACE} = ? 
+             AND {CATALOG_FIELD_TABLE_NAME} = ?
+             AND (
+                    {CATALOG_FIELD_RECORD_TYPE} = '{CATALOG_FIELD_TABLE_RECORD_TYPE}' 
+                    OR {CATALOG_FIELD_RECORD_TYPE} IS NULL
+                  )"
+        );
+
+        let args = vec![
+            Some(catalog_name),
+            Some(namespace.as_str()),
+            Some(table_name),
+        ];
+
+        self.execute(&delete, args, None).await?;
+
+        Ok(())
     }
 
-    async fn load_table(&self, _identifier: &TableIdent) -> Result<Table> {
-        todo!()
+    async fn load_table(&self, identifier: &TableIdent) -> Result<Table> {
+        if !self.table_exists(identifier).await? {
+            return no_such_table_err(identifier);
+        }
+
+        let namespace = identifier.namespace().join(".");
+        let table_name = identifier.name();
+        let catalog_name = self.name.as_str();
+
+        let query = format!(
+            "SELECT {CATALOG_FIELD_METADATA_LOCATION_PROP} FROM {CATALOG_TABLE_NAME} 
+                                    WHERE {CATALOG_FIELD_CATALOG_NAME} = ? 
+                                    AND {CATALOG_FIELD_TABLE_NAMESPACE} = ? 
+                                    AND {CATALOG_FIELD_TABLE_NAME} = ?
+                                    AND (
+                                        {CATALOG_FIELD_RECORD_TYPE} = '{CATALOG_FIELD_TABLE_RECORD_TYPE}' 
+                                    OR {CATALOG_FIELD_RECORD_TYPE} IS NULL)
+                                    LIMIT 1
+                                    "
+        );
+
+        let args = vec![
+            Some(catalog_name),
+            Some(namespace.as_str()),
+            Some(table_name),
+        ];
+
+        let query_result_rows = self.fetch_rows(&query, args).await?;
+
+        if query_result_rows.is_empty() {
+            return no_such_table_err(identifier);
+        }
+
+        let row = query_result_rows.first().unwrap();
+        let metadata_location = row
+            .try_get::<String, _>(CATALOG_FIELD_METADATA_LOCATION_PROP)
+            .map_err(from_sqlx_error)?;
+
+        let file = self.fileio.new_input(&metadata_location)?;
+        let metadata: TableMetadata = serde_json::from_slice(file.read().await?.as_ref())?;
+
+        Ok(Table::builder()
+            .file_io(self.fileio.clone())
+            .identifier(identifier.clone())
+            .metadata_location(metadata_location)
+            .metadata(metadata)
+            .build()?)
     }
 
     async fn create_table(
         &self,
-        _namespace: &NamespaceIdent,
-        _creation: TableCreation,
+        namespace: &NamespaceIdent,
+        creation: TableCreation,
     ) -> Result<Table> {
-        todo!()
+        if !self.namespace_exists(namespace).await? {
+            return no_such_namespace_err(namespace);
+        }
```

Comment:
Could you please help to check when the catalog table `iceberg_tables` is created? Because when we use JNI catalog to create a table, it will initialize the catalog `iceberg_tables` (i.e. create table `iceberg_tables`) for us.

Reply:
https://github.com/apache/iceberg/blob/40ffcb9ad42491c75e4c306fec1486e50849d03c/core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java#L144-L147

Reply:
The catalog table `iceberg_tables` will be created when `Create new sql catalog instance`